server/http.go

```diff
@@ -290,27 +437,102 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, webh
 
 	_, appErr = p.API.UpdatePost(post)
 	if appErr != nil {
-		p.API.LogWarn("Could not update the post", "err", appErr)
+		p.API.LogWarn("Could not update the post", "Error", appErr.Error())
 		http.Error(w, appErr.Error(), appErr.StatusCode)
 		return
 	}
 
 	if appErr = p.deleteMeetingPostID(meetingPostID); appErr != nil {
-		p.API.LogWarn("failed to delete db entry", "error", appErr.Error())
+		p.API.LogWarn("failed to delete db entry", "Error", appErr.Error())
 		return
 	}
 
 	w.Header().Set("Content-Type", "application/json")
 	if err := json.NewEncoder(w).Encode(post); err != nil {
-		p.API.LogWarn("failed to write response", "error", err.Error())
+		http.Error(w, "failed to write the response", http.StatusInternalServerError)
+	}
+}
+
+func (p *Plugin) slackAttachmentToUpdatePMI(currentValue, channelID string) *model.SlackAttachment {
+	apiEndPoint := fmt.Sprintf(APIToUpdatePMI, manifest.ID)
+
+	slackAttachment := model.SlackAttachment{
+		Fallback: "You can not set your preference",
+		Title:    "*Setting: Use your Personal Meeting ID*",
+		Text:     fmt.Sprintf("\n\nDo you want to use your Personal Meeting ID when starting a meeting?\n\nCurrent value: %s", currentValue),
+		Actions: []*model.PostAction{
+			{
+				Id:    Yes,
+				Name:  Yes,
+				Type:  model.PostActionTypeButton,
+				Style: "default",
+				Integration: &model.PostActionIntegration{
+					URL: apiEndPoint,
+					Context: map[string]interface{}{
+						Action: Yes,
+					},
+				},
+			},
+			{
+				Id:    No,
+				Name:  No,
+				Type:  model.PostActionTypeButton,
+				Style: "default",
+				Integration: &model.PostActionIntegration{
+					URL: apiEndPoint,
+					Context: map[string]interface{}{
+						Action: No,
+					},
+				},
+			},
+			{
+				Id:    Ask,
+				Name:  Ask,
+				Type:  model.PostActionTypeButton,
+				Style: "default",
+				Integration: &model.PostActionIntegration{
+					URL: apiEndPoint,
+					Context: map[string]interface{}{
+						Action: Ask,
+					},
+				},
+			},
+		},
+	}
+
+	return &slackAttachment
+}
+
+func (p *Plugin) updatePMI(userID string, channelID string) {
```

Comment:
This function name is confusing, because we aren't updating the PMI here. I think this should be named something like `sendUserSettingForm`

Reply:
Also, couldn't this just be done by a slash command? It would require much less code and complexity than what's currently being introduced in the PR

`/zoom settings pmi ask`

Reply:
@mickmister , Matt Birch and I had talk about this and the reasoning was somewhat 2 fold:

1. We wanted a place to be able to support additional settings in the future
2. We chose to follow the pattern that the MS Calendar plugin used which was to have the settings post to the channel and allow you to edit it them.

See [this doc](https://docs.google.com/document/d/1ddPplGPziKOMoA5HjBwRG4Omw-4J0dJ4UMP-0qZTbVw/edit#heading=h.fv9oaom1tgf4) for reference